ui/summary_test.go

```diff
@@ -164,3 +166,125 @@ func createTestMetrics() map[string]*stats.Metric {
 
 	return metrics
 }
+
+func TestSummarizeMetricsJSON(t *testing.T) {
+	metrics := createTestMetrics()
+	expected := `[{
+    "name": "",
+    "path": "",
+    "id": "d41d8cd98f00b204e9800998ecf8427e",
+    "groups": {
+        "child": {
+        "name": "child",
+        "path": "::child",
+        "id": "f41cbb53a398ec1c9fb3d33e20c9b040",
+        "groups": {},
+        "checks": {
+            "check1": {
+                "name": "check1",
+                "path": "::child::check1",
+                "id": "6289a7a06253a1c3f6137dfb25695563",
+                "passes": 5,
+                "fails": 10
+                }
+            }
+        }
+    },
+    "checks": {}
+},
+{
+    "checks": {
+        "extra": [
+            "✓ 3",
+            "✗ 0"
+        ],
+        "value": 0
+    },
+    "http_reqs": {
+        "count": 3,
+        "rate": 3
+    },
+    "my_trend": {
+        "avg": 15,
+        "max": 20,
+        "med": 15,
+        "min": 10,
+        "p(90)": 19,
+        "p(95)": 19.5
+    },
+    "vus": {
+        "extra": [
+            "min=1",
+            "max=1"
```

Comment:
as above I think this should be `{"min": 1, "max": 1}` 
Also `"extra"` should probably be missing and all the keys inside of it to be merged with the level above (`vus` in this case)

Reply:
Not sure it's doable with current design of internal metrics calculation.

It's probably better to create separate issue and make enhancement in next release.

Reply:
well .. this will mean it will break backwards compatibility so if you think this will take too much time .. than I vote for moving it to 0.27.0 
cc @na-- @imiric

Reply:
This also should IMO definitely mark failed thresholds as .. failed ones which it currently doesn't do

Reply:
Well, "breaking backwards compatibility" is a bit of a stretch considering the previous `--report` option was removed almost 2 years ago :smile:

But I agree that this conversion should be more thorough in both cases (parsing checks and extras), and perhaps nicer to use.

I'm not sure if this would be overkill or out of scope for this PR, but what do you think about `Summary` implementing `json.Marshaler` and `encoding.TextMarshaler`? This would standardize things a bit and maybe make it easier to test. It also opens the door to implementing the corresponding `*Unmarshaler`s if that ever becomes a requirement.

In any case, I'm fine with postponing this for 0.27.0, though several people have expressed their interest in this, so we might want to consider postponing 0.26.0 instead to include this.

Reply:
@imiric they interest in having a JSON report, this problem is about what's the easiest way to parse the JSON. So not a blocker for v0.26.0. We can introduce the feature, then improve it later in next release.